### PR TITLE
Disable `webhookReply` by default

### PR DESCRIPTION
**Pull request by wojpawlik**
# Description

`webhookReply` is an [endless source of issues](https://github.com/telegraf/telegraf/issues?q=webhookReply). To increase reliability (and reduce the number of "bug" reports), I suggest disabling it by default. Instead, I'd encourage people to enable it on per-call basis and document the gotchas:

1. Errors in the `webhookReply` calls cannot be reported.
2. `webhookReply` calls break ordering by resolving before bot API has a chance to process them (#1167, #1086).
3. `webhookReply` call confirms the update, which means it won't be processed again if bot crashes.

I'm keeping this a draft until I hear @KnorpelSenf opinion on some deprecations I wish to make here:

To discourage enabling `webhookReply` globally, I could deprecate https://github.com/telegraf/telegraf/blob/424f89552509f5490f096800a4843a8e255c5305/src/core/network/client.ts#L37

To avoid duplicating the docs, I could also deprecate https://github.com/telegraf/telegraf/blob/424f89552509f5490f096800a4843a8e255c5305/src/context.ts#L175-L181

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
I agree, this is a good idea. It breaks more than it helps, the performance benefit is irrelevant for small bots.

We should, however, also provide a proper way to enable it, even though it should maybe not in the most obvious place of the config. Any suggestions for how you would do this per request?

**Comment by wojpawlik** [comment]
I could leave `ctx.telegram.webhookReply` intact. Or I could deprecate it and introduce a `webhookReply` option to `callApi`, next to `signal`.

What do you think about the 2 deprecations I suggested?

**Comment by KnorpelSenf** [comment]
> What do you think about the 2 deprecations I suggested?

Ah, I was referring to those when I said

> I agree, this is a good idea.

I also like the rest, that should be implied.

Are you planning on doing another breaking release soon? If so, we can indeed postpone the decision about `ctx.telegram.webhookReply`. Otherwise, I think it makes sense to come to a conclusion about whether it actually should be removed or not, and then introduce the breaking changes with 4.0.

**Comment by wojpawlik** [comment]
I don't wish to delay 4.0 further.

**Comment by wojpawlik** [comment]
Even tho I don't plan 5.0 anytime soon; my plans currently end at 4.1.

**Comment by KnorpelSenf** [comment]
Alright, I see that. I'm fine with just deprecating them in 4.0

File: src/core/network/client.ts [hunk 1]
```diff
@@ -320,7 +321,7 @@ class ApiClient {
   /**
    * If set to `true`, first _eligible_ call will avoid performing a POST request.
    * Note that such a call:
-   * 1. cannot report errors,
+   * 1. cannot report errors or return meaningful value,
```
Comment (KnorpelSenf) [inline src/core/network/client.ts:324]: Sounds like we're talking about human values, when we actually mean a piece of memory

```suggestion
   * 1. cannot report errors or return a meaningful value,
```

</nitpick>

**Review by KnorpelSenf — APPROVED** [review]
Nice work

**Pull request merged**